### PR TITLE
Feature: Precision Mining Hightlighter

### PR DESCRIPTION
**Pull request by CuzImClicks**
## What
Adds a box around the particle that spawns from Precision Mining for better visibility.
<details>
<summary>Images</summary>

https://github.com/user-attachments/assets/33b584a8-6b6e-4b3f-8eaa-9e8173cefaab

</details>

## Changelog New Features
+ Added Precision Mining Highlighter. - Cuz_Im_Clicks
    * Draws a box over the Precision Mining particles.

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/mining/PrecisionMiningHighlight.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        if (event.type != EnumParticleTypes.CRIT) return
+        lookingAt = false
+        lastParticle = event.location
+    }
+
+    @SubscribeEvent
+    fun onRender(event: LorenzRenderWorldEvent) {
+        val p = lastParticle ?: return
+        val axisAlignedBB = p.add(x = -0.1, y = -0.1, z = -0.1).axisAlignedTo(p.clone().add(0.1, 0.1, 0.1))
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/features/mining/PrecisionMiningHighlight.kt]: This not performant, like at all. First of all cache the boundingbox second create the box directly with the primary constructor.

File: src/main/java/at/hannibal2/skyhanni/features/mining/PrecisionMiningHighlight.kt
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/features/mining/PrecisionMiningHighlight.kt:1]: Use onTick event to reset it after some cool down of no particle. Something like: ```deleteTime = 5.0.ticks.fromNow()```

**Review by Thunderblade73 — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/mining/PrecisionMiningHighlight.kt (hunk 1 above)
Comment (CuzImClicks) [reply src/main/java/at/hannibal2/skyhanni/features/mining/PrecisionMiningHighlight.kt]: good idea

**Comment by CuzImClicks** [comment]
do you think i need a check for "holding mining tool" or do we just assume crit and villager happy particles only come from precision mining while in the mining areas?

**Comment by lunaynx** [comment]
Happy villager particles are also emitted by the wishing compass. And I think egglocator. And certain runes.

Crit particles appear on treasure chests when you don't have Great Explorer 20.

**Comment by CuzImClicks** [comment]
> Happy villager particles are also emitted by the wishing compass. And I think egglocator. And certain runes.
> 
> Crit particles appear on treasure chests when you don't have Great Explorer 20.

i can check for 
- `keyBindAttack.isKeyDown`
- holding pickaxe or drill
- block that is in breaking range

but all of those would just kick the rock further down the road?

**Comment by CuzImClicks** [comment]
fuck its bugged it only works on the south face

**Review by Thunderblade73 — APPROVED** [review]

**Review by CalMWolfs — APPROVED** [review]

**Pull request merged**